tsdb/engine/tsm1/file_store.go

```diff
@@ -26,10 +26,10 @@ type TSMFile interface {
 
 	// ReadAt returns all the values in the block identified by entry.
 	ReadAt(entry *IndexEntry, values []Value) ([]Value, error)
-	ReadFloatBlockAt(entry *IndexEntry, values []FloatValue) ([]FloatValue, error)
-	ReadIntegerBlockAt(entry *IndexEntry, values []IntegerValue) ([]IntegerValue, error)
-	ReadStringBlockAt(entry *IndexEntry, values []StringValue) ([]StringValue, error)
-	ReadBooleanBlockAt(entry *IndexEntry, values []BooleanValue) ([]BooleanValue, error)
+	ReadFloatBlockAt(entry *IndexEntry, tdec TimeDecoder, vdec *FloatDecoder, values *[]FloatValue) ([]FloatValue, error)
```

Comment:
The reads need to update the original buffer reference. Otherwise we'd need to either preallocate the buffer (which is bad when selecting a few points across a lot of series) or we have to reallocate on every read (which sucks in general).